web/src/app/services/ServiceRouter.js

```diff
@@ -31,66 +32,103 @@ const query = gql`
   }
 `
 
-export default class ServiceRouter extends React.PureComponent {
-  renderList = () => (
-    <SimpleListPage
-      query={query}
-      variables={{ input: { favoritesFirst: true } }}
-      mapDataNode={n => ({
-        title: n.name,
-        subText: n.description,
-        url: n.id,
-        isFavorite: n.isFavorite,
-      })}
-      createForm={<ServiceCreateDialog />}
-      createLabel='Service'
-    />
-  )
+export default function ServiceRouter() {
+  const searchParam = useSelector(searchSelector) // current total search string on page load
+  const dispatch = useDispatch()
+  const setSearchParam = value => dispatch(setURLParam('search', value))
 
-  renderDetails = ({ match }) => (
-    <ServiceDetails serviceID={match.params.serviceID} />
-  )
-  renderAlerts = ({ match }) => (
-    <ServiceAlerts serviceID={match.params.serviceID} />
-  )
-  renderKeys = ({ match }) => (
-    <IntegrationKeyList serviceID={match.params.serviceID} />
-  )
-  renderHeartbeatMonitors = ({ match }) => (
-    <HeartbeatMonitorList serviceID={match.params.serviceID} />
-  )
-  renderLabels = ({ match }) => (
-    <ServiceLabelList serviceID={match.params.serviceID} />
-  )
+  // grab key and value from the search param, if at all
+  let key = null
+  let value = null
+  if (searchParam.includes('=')) {
+    const searchSplit = searchParam.split(/(!=|=)/)
+    key = searchSplit[0]
+    value = searchSplit[2] // [1] being != or =
+  }
 
-  render() {
+  function renderList() {
     return (
-      <Switch>
-        <Route exact path='/services' render={this.renderList} />
-        <Route
-          exact
-          path='/services/:serviceID/alerts'
-          render={this.renderAlerts}
-        />
-        <Route exact path='/services/:serviceID' render={this.renderDetails} />
-        <Route
-          exact
-          path='/services/:serviceID/integration-keys'
-          render={this.renderKeys}
-        />
-        <Route
-          exact
-          path='/services/:serviceID/heartbeat-monitors'
-          render={this.renderHeartbeatMonitors}
-        />
-        <Route
-          exact
-          path='/services/:serviceID/labels'
-          render={this.renderLabels}
-        />
-
-        <Route component={PageNotFound} />
-      </Switch>
+      <SimpleListPage
+        query={query}
+        variables={{ input: { favoritesFirst: true } }}
+        mapDataNode={n => ({
+          title: n.name,
+          subText: n.description,
+          url: n.id,
+          isFavorite: n.isFavorite,
+        })}
+        createForm={<ServiceCreateDialog />}
+        createLabel='Service'
+        searchAdornment={
+          <ServiceLabelFilterContainer
+            labelKey={key}
+            labelValue={value}
+            onKeyChange={onKeyChange}
+            onValueChange={onValueChange}
+            onReset={() => setSearchParam()}
+          />
+        }
+      />
     )
   }
+
+  function onKeyChange(newKey) {
```

Comment:
As above, if we change to an object this can be simplified like:
```js
function onChange(val) {
    setSearchParam(val.key?`${val.key}=${val.value||''}`:null)
}
```

Reply:
this might reduce the number of lines, but greatly reduces readability